tests/CppWinRTComServerTests.cpp

```diff
@@ -129,32 +132,49 @@ TEST_CASE("CppWinRTComServerTests::AnyRegisterFailureClearAllRegistrations", "[c
 
 TEST_CASE("CppWinRTComServerTests::NotifierAndRegistration", "[cppwinrt_com_server]")
 {
-    wil::unique_event moduleEvent(wil::EventOptions::None);
-    wil::unique_event coroutineRunning(wil::EventOptions::None);
-    wil::unique_event coroutineContinue(wil::EventOptions::None);
+    wil::unique_event moduleEvent(wil::EventOptions::ManualReset);
+    wil::unique_event coroutineRunning(wil::EventOptions::ManualReset);
+    wil::unique_event coroutineContinue(wil::EventOptions::ManualReset);
 
     wil::notifiable_module_lock::instance().set_notifier([&]() {
         moduleEvent.SetEvent();
     });
+    auto resetOnExit = wil::scope_exit([&] {
+        wil::notifiable_module_lock::instance().set_notifier(nullptr);
+    });
 
     winrt::init_apartment();
 
     auto revoker = wil::register_com_server<MyServer>();
 
-    [&]() -> winrt::Windows::Foundation::IAsyncAction {
-        co_await winrt::resume_background();
-        coroutineRunning.SetEvent();
+    std::exception_ptr coroutineException;
+    auto asyncLambda = [&]() -> winrt::Windows::Foundation::IAsyncAction {
+        try
+        {
+            co_await winrt::resume_background();
+            coroutineRunning.SetEvent();
 
-        coroutineContinue.wait();
-        auto instance = create_my_server_instance();
-        REQUIRE(winrt::get_module_lock() == 2);
-    }();
+            coroutineContinue.wait();
+            auto instance = create_my_server_instance();
+            REQUIRE(winrt::get_module_lock() == 2);
+        }
+        catch (...)
+        {
+            coroutineException = std::current_exception();
+        }
+    };
+    asyncLambda();
 
     coroutineRunning.wait();
     REQUIRE(winrt::get_module_lock() == 1); // Coroutine bumped count
 
     coroutineContinue.SetEvent();
     moduleEvent.wait();
 
+    if (coroutineException)
+    {
+        std::rethrow_exception(coroutineException);

```

Comment:
Out of curiosity why not just REQUIRE that the corotuneException is empty?  Does the rethrow make catch2 print something useful?